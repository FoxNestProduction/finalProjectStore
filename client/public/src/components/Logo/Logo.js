import React from 'react';

const Logo = () => {
  return (
    <div className="logo">
      <img src="./img/logo.png" alt="logo" />
      <p className="logo__text">eatly</p>
    </div>
  );
};

export default Logo;
