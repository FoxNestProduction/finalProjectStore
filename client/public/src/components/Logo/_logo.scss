@import "../../styles/mixins";
@import "../../styles/variables";

.logo {
  @include flex(center, center, row);
}
.logo__text {
  margin-left: 10px;
  font-family: $font-poppins;
  font-weight: 600;
  font-size: 18px;
  color: $primaryDefault-color;
}