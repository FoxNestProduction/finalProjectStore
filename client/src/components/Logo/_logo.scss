@import "../../styles/mixins";
@import "../../styles/variables";

.logo {
  @include flex(center, center, row);
  img {
    width: 40px;
    height: 37px;
  }

  @media (min-width: $bp-tablet) {
    img {
      width: 44px;
      height: 40px;
    }
  }

  @media (min-width: $bp-desktop) {
    img {
      width: 62px;
      height: 57px;
    }
  }
}
.logo__text {
  margin-left: 10px;
  font-family: $font-poppins;
  font-weight: 600;
  font-size: 18px;
  color: $primaryDefault-color;

  @media (min-width: $bp-tablet) {
    font-size: 20px;
  }

  @media (min-width: $bp-desktop) {
    font-size: 29px;
  }
}
