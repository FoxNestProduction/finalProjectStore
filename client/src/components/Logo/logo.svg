<svg width="41" height="38" viewBox="0 0 41 38" fill="none" xmlns="http://www.w3.org/2000/svg">
<rect x="3.25098" y="4.40625" width="37.4025" height="33.0393" rx="6.16733" fill="#6C5FBC"/>
<rect x="0.891536" y="0.891536" width="35.6194" height="31.2562" rx="5.2758" fill="#DBD9EE" stroke="#6C5FBC" stroke-width="1.78307"/>
<path d="M28.8682 10.6655L8.60258 10.5759" stroke="#6C5FBC" stroke-width="1.84442" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M28.8446 14.7226L24.344 14.709C22.4824 14.7033 20.9793 12.8782 20.9893 10.6355C20.9993 8.39284 22.5186 6.57689 24.3802 6.58254L28.8809 6.59619" stroke="#6C5FBC" stroke-width="1.84442" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M15.3583 23.6466L9.47517 23.5891C8.99312 23.5877 8.58987 23.1108 8.58151 22.5295L8.54451 20.9968C8.53627 20.389 8.94393 19.875 9.44789 19.8765L15.2653 19.9338L15.3583 23.6466Z" stroke="#6C5FBC" stroke-width="1.84442" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M24.6072 25.8539L16.2891 25.7625C15.8068 25.7611 15.4035 25.2841 15.3951 24.7025L15.2958 19.9311L28.8086 20.0778L28.8168 20.6858C28.8807 23.5672 26.9853 25.8875 24.6072 25.8539Z" stroke="#6C5FBC" stroke-width="1.84442" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
