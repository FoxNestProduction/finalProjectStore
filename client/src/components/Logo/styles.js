export const stylesIcon = {
  p: 0,
  mr: '10px',
};

export const stylesIconHeader = {
  ...stylesIcon,
  width: {
    mobile: '45px',
    lgTablet: '32px',
    desktop: '45px',
  },
  height: {
    mobile: '42px',
    lgTablet: '30px',
    desktop: '42px',
  },
};

export const stylesIconFooter = {
  ...stylesIcon,
  width: {
    mobile: '40px',
    tablet: '44px',
    desktop: '62px',
  },
  height: {
    mobile: '37px',
    tablet: '40px',
    desktop: '57px',
  },
};

export const stylesTitle = {
  color: 'primary.main',
  fontFamily: 'fontPoppins',
  fontWeight: 'fontWeightSemiBold',
};

export const stylesTitleHeader = {
  ...stylesTitle,
  fontSize: {
    mobile: '21px',
    lgTablet: '15px',
    desktop: '21px',
  },
};

export const stylesTitleFooter = {
  ...stylesTitle,
  fontSize: {
    mobile: '18px',
    tablet: '20px',
    desktop: '29px',
  },
};
