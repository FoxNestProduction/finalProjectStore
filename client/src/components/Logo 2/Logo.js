import React from 'react';
import PropTypes from 'prop-types';
import { ReactComponent as LogoIcon } from './logo.svg';

const Logo = ({ className, classNameIcon }) => {
  return (
    <div className={className}>
      <LogoIcon className={classNameIcon} />
      <p>eatly</p>
    </div>
  );
};

export default Logo;

Logo.propTypes = {
  className: PropTypes.string,
  classNameIcon: PropTypes.string,
};

Logo.defaultProps = {
  className: '',
  classNameIcon: '',
};
