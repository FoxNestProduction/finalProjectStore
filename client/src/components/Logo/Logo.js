import React from 'react';
import { IconButton, Stack, Typography } from '@mui/material';
import PropTypes from 'prop-types';
import { ReactComponent as LogoIcon } from '../../assets/svg/logo.svg';
import {
  stylesIconFooter,
  stylesIconHeader,
  stylesTitleFooter,
  stylesTitleHeader,
} from './styles';

const Logo = ({ type }) => {
  return (
    <Stack direction="row" alignItems="center">
      <IconButton
        component="div"
        edge="center"
        aria-label="logo"
        sx={type === 'header' ? stylesIconHeader : stylesIconFooter}
      >
        <LogoIcon />
      </IconButton>
      <Typography
        sx={type === 'header' ? stylesTitleHeader : stylesTitleFooter}
      >
        eatly
      </Typography>
    </Stack>
  );
};

Logo.propTypes = {
  type: PropTypes.oneOf(['header', 'footer']),
};

Logo.defaultProps = {
  type: 'header',
};

export default Logo;
