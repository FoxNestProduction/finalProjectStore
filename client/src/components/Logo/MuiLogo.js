import React from 'react';
import { IconButton, Stack, Typography } from '@mui/material';
import { ReactComponent as LogoIcon } from '../../assets/svg/logo.svg';

const MuiLogo = () => {
  const iconSize = {
    width: {
      zero: 40,
      mobile: 40,
      tabet: 44,
      desktop: 62,
    },
    height: {
      zero: 37,
      mobile: 37,
      tabet: 40,
      desktop: 57,
    },
  };

  return (
    <Stack direction="row" alignItems="center">
      <IconButton
        edge="center"
        aria-label="logo"
        sx={{
          p: 0,
          mr: '10px',
          width: {
            zero: 40,
            mobile: 40,
            tablet: 44,
            desktop: 62,
          },
          height: {
            zero: 37,
            mobile: 37,
            tablet: 40,
            desktop: 57,
          },
        }}
      >
        <LogoIcon />
      </IconButton>
      <Typography
        sx={{
          color: 'primary.main',
          fontFamily: 'fontFamily.secondary',
          fontWeight: 'fontWeight.bold',
          fontSize: {
            zero: 18,
            mobile: 18,
            tablet: 20,
            desktop: 29,
          },
        }}
      >
        eatly
      </Typography>
    </Stack>
  );
};

export default MuiLogo;
