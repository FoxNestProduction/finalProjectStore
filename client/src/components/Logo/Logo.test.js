import { render } from "@testing-library/react";
import Logo from '../Logo/Logo'

const Component = (props) => {
  return (
    <Logo />
  );
};

describe("Snapshot test", () => {
  it("should Logo in Footer render", () => {
    const { asFragment } = render(<Component type="footer" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should Logo in Header render", () => {
    const { asFragment } = render(<Component />);

    expect(asFragment()).toMatchSnapshot();
  });
});
