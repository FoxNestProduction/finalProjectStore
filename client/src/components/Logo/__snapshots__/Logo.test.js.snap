// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test should Logo in Footer render 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-1d9cypr-MuiStack-root"
  >
    <div
      aria-label="logo"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-sb6o63-MuiButtonBase-root-MuiIconButton-root"
      role="button"
      tabindex="0"
    >
      <svg>
        logo.svg
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1n20qe9-MuiTypography-root"
    >
      eatly
    </p>
  </div>
</DocumentFragment>
`;

exports[`Snapshot test should Logo in Header render 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-1d9cypr-MuiStack-root"
  >
    <div
      aria-label="logo"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-sb6o63-MuiButtonBase-root-MuiIconButton-root"
      role="button"
      tabindex="0"
    >
      <svg>
        logo.svg
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1n20qe9-MuiTypography-root"
    >
      eatly
    </p>
  </div>
</DocumentFragment>
`;
